### PR TITLE
Build nb_platform with starboard_toolchain only

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/246854012

### CODE REVIEW

File: nb/BUILD.gn [hunk 1]
```diff
@@ -37,7 +37,8 @@ static_library("nb") {
       "//starboard/common",
     ]
 
-    if (defined(has_nb_platform) && has_nb_platform) {
+    if (defined(has_nb_platform) && has_nb_platform &&
+        current_toolchain == starboard_toolchain) {
```
Comment (andrewsavage1) [inline nb/BUILD.gn:41]: I'm not sure this seems right—I'd expect this to just not build nb_platform ever, as I don't think anything depends on nb with the starboard toolchain. Better would be to just put the nb_platform into the starboard_platform if that's at all possible

Comment (niranjanyardi) [reply nb/BUILD.gn:41]: glimp_platform under starboard on ps4 has nb as a dep : it ll get built through it

Comment (andrewsavage1) [reply nb/BUILD.gn:41]: Ah okay—and are there no issues with the nb target above starboard not using nb_platform provided by the platform?

Comment (niranjanyardi) [reply nb/BUILD.gn:41]: this seems janky and should be fixed for all platforms doing this in a follow up cl - if the starboard side depends on nb/BUILD.gn 


No issues AFAIK - cobalt runs and loads fine . Maybe some tests catches something else but nothing as of now. there's no way modular builds for ps4/5 can do this as the nb platform specific stuff contains sce libraries :  and those are the only 2 platforms doing this : has_nb_platform = true is set for PS4/5 only.

Comment (niranjanyardi) [reply nb/BUILD.gn:41]: I'll add a TODO there to clean this up later

Comment (andrewsavage1) [reply nb/BUILD.gn:41]: Have you already tried putting nb_platform under starboard_platform? If that just works I think it would be cleaner

Comment (niranjanyardi) [reply nb/BUILD.gn:42]: Tried, failed, details in TODO bug

**Review by kaidokert — APPROVED** [review]

**Pull request merged**